Lib/importlib/_bootstrap_external.py

```diff
@@ -91,22 +92,35 @@ def _unpack_uint16(data):
     return int.from_bytes(data, 'little')
 
 
-def _path_join(*path_parts):
-    """Replacement for os.path.join()."""
-    return path_sep.join([part.rstrip(path_separators)
-                          for part in path_parts if part])
+if _MS_WINDOWS:
+    def _path_join(*path_parts):
+        """Replacement for os.path.join()."""
+        if not path_parts:
+            return ""
+        root = ""
+        path = []
+        for new_root, tail in map(_os._path_splitroot, path_parts):
+            if new_root.startswith(path_sep_tuple) or new_root.endswith(path_sep_tuple):
+                root = new_root.rstrip(path_separators) or root
+                path = [path_sep + tail]
+            else:
+                root = new_root or root
+                path.append(tail)
```

Comment:
It's not handling drive-relative paths at all, so the result is undefined behavior. I'd prefer to raise an exception. For example:

```python
        elif new_root:
            raise RuntimeError('drive-relative paths are not '
                               f'supported: {new_root + tail!r}')
        else:
            path.append(tail)
```

Reply:
Exceptions from here may just crash and not get displayed, so we need to do something more reasonable.

Reply:
Ended up leaving them in, so any drive-relative components (that aren't overridden by another absolute component) will result in a drive-relative path.